Always check if upgrading to the allowed RHEL version

bocekm opened the pull request:
Running the Preupgrade Assistant is a mandatory prerequisite before upgrading the system with the Red Hat Upgrade Tool. The Preupgrade Assistant performs a set of checks and many of them are relevant for one specific upgrade "path" only, e.g. for upgrades of RHEL 6.9 to RHEL 7.4. Because the Red Hat Upgrade Tool works with the information provided by the Preupgrade Assistant, the Red Hat Upgrade Tool needs to allow upgrading the system **only** to such a RHEL version for which the Preupgrade Assistant analysis has been designed for.

There was a limited[1] check just for _network_ as a source; for _iso_ or _device_ sources there wasn't any check at all. Now, the RHEL version specified in the _.treeinfo_ file of the installation media (be it network, iso or device) need to match the RHEL version specified by the Preupgrade Assistant.

This PR also includes:
- rewording some of the user-facing messages
- code cleanup: spaghetti into smaller functions

[1] The limited check was comparing the RHEL version passed through the `--network` option against the Preupgrade Assistant-generated data.

Resolves: https://github.com/upgrades-migrations/redhat-upgrade-tool/issues/30

bocekm committed:
Reword user-facing messages

bocekm committed:
Code cleanup: spaghetti into small functions

bocekm committed:
Replace use of undefined message() function

bocekm committed:
Always check if upgrading to the latest RHEL 7

- there was such a check, but just for network as a source; for iso or dev
  sources this check wasn't there

pirat89 commented:
Code seems OK, I will test it during monday.

jhornice commented on redhat-upgrade-tool.py:
```diff
@@ -103,31 +107,37 @@ def transaction_test(pkgs):
+        raise SystemExit(1)
+    installation_media_sysver = treeinfo.get('general', 'version')
+    if preupg_supported_sysver != installation_media_sysver:
+        print _("The installed version of Preupgrade Assistant allows upgrade"
```
the Preupgrade Assistant

jhornice commented on redhat-upgrade-tool.py:
```diff
@@ -155,97 +165,20 @@ def main(args):
+        check_same_variant_upgrade(f.treeinfo)
     else:
-        log.info("Skipping variant check")
+        log.info("Skipping system variant check.")
```
Skipping the system variant check.

jhornice commented on redhat-upgrade-tool.py:
```diff
@@ -326,7 +258,7 @@ def handle_sigwinch(signum, frame):
     f.save_repo_configs()
 
-    #dump all configuration to upgrade.conf, other tools need to know
+    # Dump all configuration to upgrade.conf, other tools need to know
```
I do not understand

bocekm replied:
This change (space after # and start with capital letter) is really just formal and was not necessary. But I did it:
- to follow PEP formating (https://www.python.org/dev/peps/pep-0008/#block-comments)
- to have the comment consistent with the other comments

artmello reviewed (commented):
code lgtm

bocekm committed:
Update man/help description of the --network option

- the value of the --network option now does not serve any other purpose
  than replacing the $releasever variable in any repo URL used by the
  Red Hat Upgrade Tool

pirat89 commented on redhat_upgrade_tool/commandline.py:
```diff
@@ -85,9 +85,9 @@ def parse_args(gui=False):
     req.add_option('--network', metavar=_('VERSION'), type="VERSION",
-        help=_('online repos matching VERSION (a number or "rawhide")'))
+        help=_('online repos. \'RELEASEVER\' will be used to replace'
+               ' &releasever variable should it occur in any repo URL.'))
```
Should be *ampersand* really here ?

bocekm replied:
Correct, it should be dollar sign. Thanks.

pirat89 commented:
Just waiting for answer before approval (the solutions works as expected).

bocekm commented:
@jhornice, please check the English of the latest commit changes. Thanks.

pirat89 reviewed (dismissed):
Giving approve, so it can be merged after the verification by @jhornice when there it is not error :-)

pkotvan commented:
This is pretty easy to reproduce and verify. Just need to try to upgrade to some older minor version of RHEL-7. Ideally using at least a network source and iso.

pkotvan commented:
When I tried to verify this I encountered this traceback when calling `redhat-upgrade-tool`:

```
# redhat-upgrade-tool                        
Traceback (most recent call last):                                  
  File "/usr/bin/redhat-upgrade-tool", line 35, in <module>         
    from redhat_upgrade_tool.commandline import parse_args, do_cleanup, device_setup                                                    
  File "/usr/lib/python2.6/site-packages/redhat_upgrade_tool/commandline.py", line 238, in <module>                                     
    class Option(optparse.Option):                                  
  File "/usr/lib/python2.6/site-packages/redhat_upgrade_tool/commandline.py", line 244, in Option                                       
    TYPE_CHECKER["VERSION"] = VERSION                               
NameError: name 'VERSION' is not defined
```

I'm not sure if this is me doing something wrong, or this is caused by this change.

bocekm committed:
Language check of the man page

pkotvan commented:
I was able to verify the fix. Upgrade attempt from RHEL-6.9 to RHEL-7.3 using `--iso` and `--network` both exitting with following message *The installed version of Preupgrade Assistant allows upgrade only to the system version 7.4.*
